octopusdeploy_framework/resource_tag.go

```diff
@@ -146,15 +147,20 @@ func (t *tagTypeResource) Update(ctx context.Context, req resource.UpdateRequest
 		}
 
 		tag := schemas.MapFromStateToTag(data)
+		if tag.ID == "" {
+			tag.ID = destinationTagSet.GetID() + "/" + strings.Split(state.ID.ValueString(), "/")[1]
```

Comment:
This is the actual fix, setting the tag ID in the configuration itself does not handle other scenarios correctly.